From: Simon McVittie <smcv@debian.org>
Date: Mon, 18 Dec 2017 10:47:29 +0000
Subject: closures test: Run fewer iterations on ARM64 and armhf

It appears that this feature does *work* on ARM64 and armhf, it's just
really slow. Cut down the number of signal emission cycles so that
builds won't time out while the root cause is investigated.

Updated by Iain Lane <laney@debian.org>: some of our armhf buildds are
now on arm64 hardware, and also fail in this way.

Forwarded: no, Debian-specific hack
Bug: https://gitlab.gnome.org/GNOME/glib/issues/1316
Bug-Debian: https://bugs.debian.org/880883
Last-Updated: 2018-09-07
Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tests/refcount/closures.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/refcount/closures.c b/tests/refcount/closures.c
index 25b0a27..838c45e 100644
--- a/tests/refcount/closures.c
+++ b/tests/refcount/closures.c
@@ -259,7 +259,11 @@ main (int    argc,
   thread1 = g_thread_create (thread1_main, closure, TRUE, NULL);
   thread2 = g_thread_create (thread2_main, closure, TRUE, NULL);
 
+#if defined(__aarch64__) || defined(__arm__)
+  for (i = 0; i < 100000; i++)
+#else
   for (i = 0; i < 1000000; i++)
+#endif
     {
       static guint count = 0;
       test_emissions (object);
